### PR TITLE
Feature - option to copy reports into variant-specific asset directories

### PR DESCRIPTION
**Pull request by rabidaudio**
I appreciate the plugin!

I noticed that the `copy*ReportToAssets` options put the assets in the `main` assets folder, but we have multiple flavors in our app so to keep them from overwriting each other I added an option `useVariantSpecificAssetDirs` that puts them in the flavor-specific asset folder. I could see this as a first step to a `licenseReport` task that would generate reports for all variants. I'm not sold on the option name and open to changes to the api or implementation. Also totally understand if this is a feature you don't want to support.

I'm not sure how good the test is, I'm not very familiar with the test framework.

### CODE REVIEW

**Comment by jaredsburrows** [comment]
Can you resolve conflicts so we can clean up and merge?

**Comment by rabidaudio** [comment]
Updated to the new project code. I've tested it in our project with jitpack and it works as expected.

**Comment by jaredsburrows** [comment]
@rabidaudio Sorry for late reply, can you sync this and we can merge it?

**Comment by rabidaudio** [comment]
@jaredsburrows sync'd!

File: gradle-license-plugin/src/main/kotlin/com/jaredsburrows/license/projectAndroid.kt [hunk 1]
```diff
@@ -89,9 +89,10 @@ private fun Project.configureVariant(
     }
 
     tasks.register("license${name}Report", LicenseReportTask::class.java) {
+      val sourceSetName = if (it.useVariantSpecificAssetDirs) variant.name else "main"
       it.assetDirs = baseExtension
         .sourceSets
-        .getByName("main")
+        .getByName(sourceSetName)
```
Comment (jaredsburrows) [inline gradle-license-plugin/src/main/kotlin/com/jaredsburrows/license/projectAndroid.kt:95]: Thanks!

**Comment by jaredsburrows** [comment]
Thanks!

**Pull request merged**